### PR TITLE
Align with sql like readme

### PR DESCRIPTION
**Pull request by mendelB**
In the Readme lesson for this Lab we have changed our has_many approach to be more aligned with the way sql does it and less redundant.
Now an Artist will have_many songs through a songs method that selects all songs with an artist equal to self for the Songs classes all array - a list of all created songs.

The readme and specs have been changed to reflect this.
- Removed the tests for artist class to count the songs. Previously an artist instance was tasked with keeping track of all its own songs so an overall class counter may have made more sense. Now these concerns are separated and such a counter is irrelevant.
- Removed the add_song method from the artist. Same idea here, the artist is associated with the song by being assigned to it, or passed in as an argument on creation. there is no need for it to take a song and assign self as the song artist when we can merely write `song.artist = artist`.

All of the above applies to the Author Post relationship in this lab.

This is relevant to [this issue](https://github.com/learn-co-curriculum/ruby-objects-has-many-readme/issues/30) and [this pull request](https://github.com/learn-co-curriculum/ruby-objects-has-many-readme/pull/31)

NOTE: Not to be merged until the entire section had been modified to align with this approach

I would love feedback/criticism/a-sound-thrashing-cuz-i-messed-this-up on this.

**cc**: @pletcher @AnnJohn @aturkewi @PeterBell

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,34 +13,31 @@ In this lab, we'll be dealing with two sets of classes. The `Artist` and `Song`
...
+How can an artist have many songs and a song belong to an artist? 
 
-How can an author have many posts and a post belong to an author? An individual post will need to have an `author=()` and `author` method and an individual author will need to be initialized with an empty collection of posts.
+An individual song will need to be able to be initialized with an `artist` object -- which can be passed in as an argument to the Song.new method -- and should have an `#artist` getter method to reveal the artist. The Song class itself should have a class variable `@@all` equal to an array which will contain each created Song, the songs should be added to that array when they are created. It should be able to reveal that array of songs with an `.all` class method.
 
-#### The `#add_song_by_name` and `#add_post_by_title` Methods
+An individual artist will need to have a `#songs` method that will run a `select` iteration on the return value of the Song classes `all` method (The array of songs contained in its `@@all` array), and select the songs whose artist is equal to `self`.
```
Comment (pletcher) [inline README.md]: "Song classes `all` method" -> "Song class' `all` method"

"(The array of songs" -> "(the array of songs"

File: README.md [hunk 2]
```diff
@@ -13,34 +13,31 @@ In this lab, we'll be dealing with two sets of classes. The `Artist` and `Song`
...
+
+How can an author have many posts and a post belong to an author? Well by using the same exact logic we used for our Song and Artist classes! The Post class will keep track of all created posts in an `@@all` class variable, exposed by it's `all` getter method. And each instance of post will hold on to it's author. An author will be able to find it's posts by calling the Post classes `all` method, and running a `select` iteration on that array to capture each post with an author equal to self.
```
Comment (pletcher) [inline README.md:32]: We could use a stronger transition between the previous paragraph and this one.

File: README.md [hunk 3]
```diff
@@ -13,34 +13,31 @@ In this lab, we'll be dealing with two sets of classes. The `Artist` and `Song`
...
 
-To get the tests on these methods passing, you'll need to define methods that take in an argument of a name (or title), use that argument to create a new song (or post) and *then* associate the objects. 
+#### The `#add_song_by_name` and `#add_post_by_title` Methods
 
-For `add_post_by_title`, you'll want to make sure the body of the method associates the newly created post with an author and adds the post to the author's collection.
+To get the tests on these methods passing, you'll need to define methods that take in an argument of a name (or title), and use that argument to create a new song (or post) passing in `self` as the artist (or author) which will associate the objects to each other. 
 
 #### The `#artist_name` and `#author_name` Methods
 
 Build the methods `some_song.artist_name` and `some_post.author_name` that return the names of the artist and author of the given song and post respectively.
```
Comment (pletcher) [inline README.md]: I know this isn't part of the changes, but we should use Ruby's documentation syntax here: `Song#artist_name` and `Post#author_name`.

**Comment by pletcher** [comment]
I like this! I left a couple of minor suggestions :)

**Comment by AnnJohn** [comment]
@mendelB can you look over and update? thanks!

**Comment by mendelB** [comment]
@AnnJohn Updated with https://github.com/learn-co-curriculum/ruby-objects-has-many-lab/pull/6/commits/2a661a054986482169fd9568c5ee41ae2eb78d39

File: spec/artist_spec.rb [hunk 4]
```diff
@@ -3,53 +3,39 @@
...
-      it "takes in an argument of a song name, creates a new song with it and associates the song and artist" do 
-        adele.add_song_by_name("Rolling in the Deep")
-        expect(adele.songs.last.name).to eq("Rolling in the Deep")
-        expect(adele.songs.last.artist).to eq(adele)
-      end
+    it "does not select the songs which don't 'belong' to itself" do
+      nintynine_problems = Song.new("Ninty Nine Problems")
+      expect(adele.songs).to be_a(Array)
+      expect(adele.songs).to include(hello)
+      expect(adele.songs).to_not include(nintynine_problems)
```
Comment (pletcher) [inline spec/artist_spec.rb:30]: If only, if only

**Comment by PeterBell** [comment]
Hey @mendelB is this good? If so, any way to resolve conflicts and merge? Ping me if qns.

**Comment by DanielSeehausen** [comment]
deprecated

**Pull request closed without merging**